R/prepare_geo_data.R

```diff
@@ -52,23 +54,90 @@ prepare_geo_data <- function(data,
     inner_join(nuts_de, by = "geo")
 
   # merge shapefile with financial data
-  financial_geo <- data |>
+  geo <- data |>
     filter(benchmark == benchmark_arg) |>
     left_join(shp_1, by = "postcode") |>
     st_as_sf()
 
-  # Add the code for data aggregation and color mapping
-  aggregated_data <- financial_geo |>
-    group_by(.data$postcode, .data$xctr_risk_category) |>
-    summarize(count = n()) |>
-    group_by(.data$postcode) |>
-    mutate(proportion = .data$count / sum(.data$count)) |>
-    ungroup()
+  aggregated_data <- aggregate_geo_data(geo, mode)
+
+  return(list(shp_1, aggregated_data))
+}
+
+
+#' Aggregate Geo Data
+#'
+#' @param geo A data frame containing geographical data.
+#' @param mode The mode to plot. It can be one of "equal_weight", "worst_case"
+#' or "best_case". If nothing is chosen, "equal_weight" is the default mode.
+#'
+#' @return A data frame with aggregated data, with the colors proportional to
+#' the risks.
+#' @noRd
+#'
+#' @examples
+#' library(tibble)
+#'
+#' # Create a sample geo_data tibble
+#' geo_data <- tibble(
+#'   postcode = c("1", "2", "3"),
+#'   company_name = c("A", "B", "C"),
+#'   risk_category_var = factor(c("low", "medium", "high"),
+#'     levels = c("low", "medium", "high")
+#'   )
+#' )
+#'
+#' aggregated_data <- aggregate_geo_data(geo_data, mode = "worst_case")
```

Comment:
The variable is useful only if you use it later. Otherwise it's best to not assign the result to anything so it's easier to see the result when we run the example.

Reply:
So the example is unnecessary here ?

Reply:
R CMD check doesn't enforce examples in internal functions but you may decide it's useful to future-self or other developers. Totally up to you.

My point was a different one though. In general terms, this reprex makes my point concrete:

``` r
library(dplyr, warn.conflicts = FALSE)

data <- tibble(x = 1)

# Defining a variable here is useful because it's used again
filtered <- filter(data)
filtered |> anyNA()
#> [1] FALSE

# Defining a variable here is not useful because:
# * It's not used again
# * The funciton name sufficiently explains what the output is
# * Running this line does not print the output :-(
filtered <- filter(data)

# So it's best to remove the variable. You loose nothing and gain printed output
filter(data)
#> # A tibble: 1 × 1
#>       x
#>   <dbl>
#> 1     1
```

<sup>Created on 2024-02-13 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>